### PR TITLE
refactor: enable most macOS funcs for apple_targets

### LINKED ISSUE #2238: Enable most macos functions on all of apple_targets
A quick search through libc shows that there's only one function we use that's defined on macos but isn't defined on tvos, watchos, and ios as well: `clock_settime`.  Even that, I suspect, is a mistake, given that `clock_gettime` is defined for all of them.
We should replace `#[cfg(target_os = macos)]` with `#[cfg(apple_targets)]` everywhere.
Honestly, I don't know why Apple needs so many different Rust targets.

### PR DESCRIPTION
**Pull request by SteveLauC**
## What does this PR do

1. Enable most macOS functions for apple_targets
2. Enable `ClockId::setime()` and `time::clock_settime()` on macOS

Closes #2238

## Checklist:

- [x] I have read `CONTRIBUTING.md`
- [ ] I have written necessary tests and rustdoc comments
- [x] A change log has been added if this PR modifies nix's API

### CODE REVIEW

File: src/time.rs [hunk 1]
```diff
@@ -40,7 +40,11 @@ impl ClockId {
     }
 
     /// Sets time to `timespec` on the clock id
-    #[cfg(not(any(apple_targets, target_os = "redox", target_os = "hermit",)))]
+    #[cfg(not(any(
```
Comment (SteveLauC) [inline src/time.rs:43]: Enable it only on macOS

File: changelog/2241.changed.md [hunk 2]
```diff
@@ -0,0 +1 @@
+`ClockId::set_time()` has been disabled on iOS/tvOS/watchOS given that it is only available on macOS
```
Comment (asomers) [inline changelog/2241.changed.md:1]: It was already disabled on iOS/tvOS/watchOS.  Shouldn't this read "ClockId::set_time is now enabled on macos" ?

File: src/time.rs [hunk 3]
```diff
@@ -40,7 +40,11 @@ impl ClockId {
...
+        target_os = "macos",
```
Comment (asomers) [inline src/time.rs:44]: Did you not notice the `not` ?
```suggestion
    #[cfg(not(any(
        target_os = "ios",
        target_os = "tvos",
        target_os = "watchos",
```

**Review by asomers — CHANGES_REQUESTED** [review]

File: src/time.rs (hunk 3 above)
Comment (SteveLauC) [reply src/time.rs]: Oops, I am sorry I missed that, fixed!

**Review by asomers — APPROVED** [review]

**Pull request merged**